app/src/main/java/com/android/nikhil/worldnow/ui/news/NewsActivity.kt

```diff
@@ -1,50 +1,38 @@
 package com.android.nikhil.worldnow.ui.news
 
 import android.arch.lifecycle.Observer
-import android.content.Intent
-import android.net.Uri
 import android.os.Bundle
-import android.support.v7.widget.LinearLayoutManager
 import android.util.Log
 import com.android.nikhil.worldnow.R
-import com.android.nikhil.worldnow.utils.NewsItemClickListener
 import com.android.nikhil.worldnow.ui.base.BaseActivity
-import com.android.nikhil.worldnow.utils.Result
-import kotlinx.android.synthetic.main.activity_main.newsRecyclerView
+import kotlinx.android.synthetic.main.activity_main.*
 import javax.inject.Inject
 
-class NewsActivity : BaseActivity<NewsViewModal>(), NewsItemClickListener {
+class NewsActivity : BaseActivity<NewsViewModal>() {
 
-    override fun getViewModelClass(): Class<NewsViewModal> = NewsViewModal::class.java
-
-    private val TAG = NewsActivity::class.java.simpleName
     private lateinit var adapter: NewsRecyclerAdapter
     @Inject lateinit var mViewModel: NewsViewModal
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
 
-        // Get the ViewHolder for this activity
-        adapter = NewsRecyclerAdapter(this, this, ArrayList<Result>())
+        adapter = NewsRecyclerAdapter(this, ArrayList())
         newsRecyclerView.adapter = adapter
-        newsRecyclerView.layoutManager = LinearLayoutManager(this)
 
-        // Observe the MutableLiveData in the ViewHolder.
         mViewModel.getNewsData().observe(this, Observer { list ->
             run {
-                if (list != null) adapter.swapNewsData(list)
-                else Log.d(TAG, "Updated list is null")
+                if (list != null) {
+                    adapter.swapNewsData(list)
```

Comment:
It will be good if after the data has been received, then the `newsListProgressBar` is hidden.

Reply:
I actually added the  line to hide it but then removed it. I'll add it and then delete the list_item.xml file as well if there are no other issues in this PR.

Reply:
@nikhilbansal97 done!